### DISCUSSION TITLE
What is a `stdint.h` and why does it included in Xcode target?

### DISCUSSION THREAD

**Discussion by lolgear** [libgit2/libgit2#5971] [Q&A]
I guess that this header should be excluded from Xcode target and ( of course ) it should be included only for win32 targets.
Maybe it is a legacy, so, it can be removed.

**Comment by ethomson** [comment]
It's not legacy, it's a C99 standard.  https://pubs.opengroup.org/onlinepubs/009604599/basedefs/stdint.h.html

Are you having trouble with it?

**Reply by lolgear** [reply]
I mean "include/git2/stdint.h".
It is included in Xcode target.

**Reply by lolgear** [reply]
I try to figure out what is going on since I started to configure a Package.swift for libgit2. This header is included and SPM tries to process it ( and hits `error` directive at the top of this header ).

**Reply by ethomson** [reply]
Oh!  Oh my.

Well, if I'm honest, I'm not sure what that's doing there.  I fear that I'm going to have to dig up some old versions of Visual Studio to try to figure out the intent by that change and if that file is really needed anymore.

**Reply by NattyNarwhal** [reply]
Just to let you know I'm using the C99 shims with a C89 compiler for my own purposes, so I'd rather it not go away.

**Reply by ethomson** [reply]
@NattyNarwhal Excellent! I'm glad to see a use case for this.  You're using an old MSVC and it needs things like `uint64_t` - is that correct?

**Reply by NattyNarwhal** [reply]
Yes, I am, and yes, it does. I have a [branch](https://github.com/visualgitio/libgit2/tree/legacy-windows) that does address compatibility issues with both old MSVC as well as backporting for older Windows. I've tried to make it clean as possible, so I should probably upstream it...

**Reply by lolgear** [reply]
We can revert an availability condition for this header, so, it will never be errored.
Embracing content of this header into `ifdef _MS_VER ... endif`.
